src/Components/FilterControls.js

```diff
@@ -1,86 +1,101 @@
 import React, { Component } from 'react';
 import CardContainer from './CardContainer';
+import '../FilterControls.css';
 
 class FilterControls extends Component {
   constructor(props) {
     super(props);
     this.state = {
       location: '',
       time: '',
-      date: ''
+      date: '',
+      matchedIds: []
     }
   }
 
-  getClickedLocation = (event) => {
-    let selectedLocation = event.target.value;
+  getClickedValue = (event) => {
     this.setState({
-      location: selectedLocation
-    }, this.filterByLocation)
+      [event.target.id]: event.target.value
+    }, this.filterParades)
   }
 
-  
-  filterByLocation = () => {
-    return this.props.parades.filter(parade => {
-      return parade.location === this.state.location
-    })
-  }
-  
-  getClickedTime = (event) => {
-    let selectedTime = event.target.value;
-    this.setState({
-      time: selectedTime
-    }, this.filterByTime)
+  filterParades = () => { 
+    const { location, time, date } = this.state;
+    const { parades } = this.props;
+    let matchedIds = [];
+    let matchedObjects = this.returnMatchingObjects();
+
+    parades.forEach(parade => {
+      if (!matchedObjects.includes(parade) && 
+      parade.location === location || 
+      parade.dayOrNight === time ||
+      parade.date === date)
+      {
+        matchedObjects.push(parade);
+        matchedIds.push(parade.id);
+        this.setState({ matchedIds });
+      }
+    });
   }
-  
-  filterByTime = () => {
-    return this.props.parades.filter(parade => {
-      return parade.dayOrNight === this.state.time
+
+  returnMatchingObjects = () => {
+    return this.props.parades.reduce((acc, parade) => {
+      this.state.matchedIds.forEach(id => {
+        if (parade.id === id) {
+          acc.push(parade)
+          console.log(parade)
+        }
     })
+      return acc
+    }, [])
```

Comment:
Might a `filter` work better here? If we filtered `this.props.parades` and our condition was instead:

`return this.state.matchedIds.includes(parade.id)` we should end up with a subset of the parades array that includes only parades that match an ID found in `this.state.matchedIds`